commons/all/src/main/java/org/infinispan/commons/configuration/io/xml/XmlConfigurationWriter.java

```diff
@@ -234,14 +234,29 @@ private void writeAttribute(String name, String value, boolean rename) {
          writer.write(rename ? naming.convert(name) : name);
          writer.write("=\"");
          if (value != null) {
-            writer.write(value.replaceAll("&", "&amp;"));
+            writer.write(escapeXML(value));
          }
          writer.write('"');
       } catch (IOException e) {
          throw new ConfigurationWriterException(e);
       }
    }
 
+   private static String escapeXML(String s) {
+      StringBuilder sb = new StringBuilder(s.length());
+      for (int i = 0; i < s.length(); i++) {
+         char ch = s.charAt(i);
+         switch (ch) {
+            case '\'' -> sb.append("&#39;");
+            case '"' -> sb.append("&#34;");
+            case '&' -> sb.append("&#38;");
+            case '<' -> sb.append("&#60;");
+            default -> sb.append(ch);
```

Comment:
Seems it is missing `>` to `&#62;`.

Reply:
Not strictly necessary according to https://www.w3.org/TR/xml/#NT-AttValue but I've added it anyway. I've also used the same logic as the YAML reader to avoid allocating a `StringBuilder` if unnecessary